### DISCUSSION TITLE
[Adonis 4.1] How to make an Adonis 4.1 app restart itself in dev mode?

### DISCUSSION THREAD

**Discussion by wemersonrv** [adonisjs/core#2231] [Help] [answered]
There is a way to restart a Adonis 4.1 application running in dev mode ?...

In production is easy, i just restart it using child_process exec() function to run a shell command... as below: 

```js
const exec = require('child_process').exec

exec('pm2 restart all', function (error, stdout, stderr) {})
```

But i need to make it in a development environment when app is run over cli command: `adonis serve --dev`

**Comment by thetutlage** [comment, marked as answer]
Not sure about the use case you have. But the `serve --dev` command restarts on file change.

Also, if I am not wrong. `pm2` also kills and starts the process again. So if want to achieve the same result, you can use pm2 in development too

**Reply by wemersonrv** [reply]
Just need to test. I think i will just detect the NODE_ENV value and print a text in console when is not in production. So i in this case i just need to save any file and it will restarts... And in production, well in this case use tne pm2 restart.

Tx @thetutlage
